### PR TITLE
Add phonetic similarity calculations

### PR DESCRIPTION
**Pull request by marcusway**
For your health!

```
>>> from similarity import SimilarityMatrix
>>> sim = SimilarityMatrix(phoneme_dict)
>>> sim.phrase_phonetic_similarity("disagree and commit", "disagree and commit")
32.42766492172225
>>> sim.phrase_phonetic_similarity("disagree and commit", "dis a green comet")
22.740264418588417
>>> sim.phrase_phonetic_similarity("disagree and commit", "poop")
-15.746563690775181
```

### CODE REVIEW

File: puns/phoneme_dictionary.py [hunk 1]
```diff
@@ -3,43 +3,53 @@
...
-				self._add(phoneme, word)
+        with open(phoneme_dict_file) as f:
+            for line in f:
+                if line.startswith(";;;"):
+                    continue
 
+                match = pattern.match(line)
+                word = self._normalize_word(match.group("word"))
+                if word is None:
+                    print line
```
Comment (randypiper) [inline puns/phoneme_dictionary.py]: Seems a little unnecessary, in my humble opinion

File: puns/similarity.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+Methods are based on those described by Hixon, Schneider, and Epstein in
+https://homes.cs.washington.edu/~bhixon/papers/phonemic_similarity_metrics_Interspeech_2011.pdf
+"""
+from math import log
+from itertools import combinations, product
+from collections import Counter, defaultdict
+
+class SimilarityMatrix(object):
+    """
+    Shit for calculating similarity
```
Comment (randypiper) [inline puns/similarity.py]: Ahem

File: puns/similarity.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                  for w in phrase.split()]
+        return [" ".join(phone_seq).split() for phone_seq in product(*phones)]
+
+
+def levenshtein(left, right):
+    """1 if values are equal else -1"""
+    return 1 if left == right else -1
+
+
+def get_aligments(seq1, seq2, gap_penalty=-1, similarity_func=levenshtein):
```
Comment (randypiper) [inline puns/similarity.py]: "alignments", perhaps?

File: puns/phoneme_dictionary.py [hunk 4]
```diff
@@ -3,43 +3,53 @@
...
 class PhonemeDictionary:
 
-	def __init__(self, phoneme_dict_file):
-		self.phonemes_to_words = collections.defaultdict(set)
-		self.words_to_phonemes = collections.defaultdict(set)
-		self._parse_phoneme_dict_file(phoneme_dict_file)
+    def __init__(self, phoneme_dict_file):
+        self.phonemes_to_words = collections.defaultdict(set)
+        self.words_to_phonemes = collections.defaultdict(set)
+        self._phonemes = set()
```
Comment (randypiper) [inline puns/phoneme_dictionary.py:9]: You can actually just grab the set of phonemes here: http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b.phones

File: puns/similarity.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
+"""
+Includes functions for computing phonetic similarity.
+Methods are based on those described by Hixon, Schneider, and Epstein in
+https://homes.cs.washington.edu/~bhixon/papers/phonemic_similarity_metrics_Interspeech_2011.pdf
```
Comment (randypiper) [inline puns/similarity.py:4]: Nice citation!

File: puns/phoneme_dictionary.py (hunk 1 above)
Comment (marcusway) [reply puns/phoneme_dictionary.py]: whoopsies

File: puns/similarity.py (hunk 2 above)
Comment (marcusway) [reply puns/similarity.py]: whoopsies

File: puns/similarity.py (hunk 3 above)
Comment (marcusway) [reply puns/similarity.py]: whoopsies

File: puns/phoneme_dictionary.py (hunk 4 above)
Comment (marcusway) [reply puns/phoneme_dictionary.py:9]: bird up!

**Pull request merged**